### PR TITLE
Move from either -> x-eithert

### PR DESCRIPTION
**Pull request by jystic**
Not sure if this is good to go yet, just wanted to use the build bots for sandboxing

### CODE REVIEW

**Comment by jystic** [comment]
I think this is all good. Just need to fix up the imports of the files I touched :)

**Comment by nhibberd** [comment]
Seems legit :trophy:

**Comment by charleso** [comment]
Death to `EitherT`, long live `EitherT`

File: mismi-s3/src/Mismi/S3/Commands.hs [hunk 1]
```diff
@@ -53,57 +54,89 @@ module Mismi.S3.Commands (
...
-import           System.IO
-import           System.Directory
-import           System.FilePath hiding ((</>))
-import           System.Posix.IO
+import           System.IO (IO, IOMode (..), SeekMode (..))
+import           System.IO (hFileSize, hSetFileSize, withFile)
+import           System.Directory (createDirectoryIfMissing, doesFileExist)
+import           System.FilePath (FilePath, dropFileName)
+import           System.Posix.IO (OpenMode(..), openFd, closeFd, fdSeek, defaultFileFlags)
 import qualified "unix-bytestring" System.Posix.IO.ByteString as UBS
```
Comment (jystic) [inline mismi-s3/src/Mismi/S3/Commands.hs]: Is this package import really really required? The current version of `unix` has the function [fdWrite](http://hackage.haskell.org/package/unix-2.7.1.0/docs/System-Posix-IO-ByteString.html#v:fdWrite) which is the only thing it is used for.

Comment (jystic) [inline mismi-s3/src/Mismi/S3/Commands.hs]: Ignore me, I see that it is a `String` for some reason :/

**Pull request merged**